### PR TITLE
Resolving Issues

### PR DESCRIPTION
**Pull request by maayarosama**
### Notes 
Create your own weblet section still needs some modifications.

### Related Issues

#1 
#2 
#3 
#4

### CODE REVIEW

File: tutorial.md [hunk 1]
```diff
@@ -1,10 +1,168 @@
 
 ### What are web components?
```
Comment (xmonader) [inline tutorial.md:2]: Can we please add an intro text of what is being explained in this tutorial what are the goals :) ?

File: tutorial.md [hunk 2]
```diff
@@ -1,10 +1,168 @@
...
+    <cover-letter></cover-letter>
+
+    ```
 
 ### [Svelte](https://svelte.dev/)
 
-Svelte is a modern JavaScript framework used to build static web applications. Users use Svelte to build reusable components for any type of projects.
+Svelte is a modern JavaScript framework used to build fast web applications. Users use Svelte to either build the entire app woth it or build reusable components that can work as standalone packages for any type of projects. 
+
+In a svelete projects it usually consists of one or more components. Each components is a reusable self-contained block with its HTML, CSS and JaveScript/Typescript encapsulated.
```
Comment (xmonader) [inline tutorial.md]: typo: svelete -> svelte

File: tutorial.md [hunk 3]
```diff
@@ -1,10 +1,168 @@
...
+    <Incrementer/>
+    <Decrementer/>
+    <Resetter/>	-
+    </button>
+    ``` 
+
+    Result:
+    ![](img/stores.gif)
 
 One of the main reason svelte is used in developing our [weblets](https://play.grid.tf/#/) is components can be used as standalone packages that work anywhere.
```
Comment (xmonader) [inline tutorial.md]: is components -> is

File: tutorial.md [hunk 4]
```diff
@@ -42,7 +199,7 @@ In our case, Pasty already has a Dockerfile in its repository. We need to add so
 ```bash
 RUN mkdir ./file_name/
 ```
-2. We need to create a new image in ubuntu based image that supports ssh server and run our pasty image in it.
+2. We need to create a new image in ubuntu based image that supports ssh server and run our pasty image in it.(inserts details here)
```
Comment (xmonader) [inline tutorial.md]: (inserts details here) is still missing

File: tutorial.md [hunk 5]
```diff
@@ -62,11 +219,468 @@ docker push image-name
...
+Next step is to install dependencies:
+```bash
+yarn
+```
+
+Now that the configuration is all set up, we need to prepare our classes and depoloyment files.
+
+Create a new folder ```types``` and cd to it.
+
+Add a ```Network.ts``` file:
```
Comment (xmonader) [inline tutorial.md:392]: please explain the purpose of this file (e.g describe the functions provided)

File: tutorial.md [hunk 6]
```diff
@@ -62,11 +219,468 @@ docker push image-name
...
+  
+    public get valid(): boolean {
+      const { name, ipRange } = this;
+      return name !== "" &&
+        ipRange !== "" 
+       
+    }
+  }
+  ```
+Add a ```rootFs.ts``` file:
```
Comment (xmonader) [inline tutorial.md:412]: please explain this file too

File: tutorial.md [hunk 7]
```diff
@@ -1,10 +1,168 @@
 
 ### What are web components?
 Web Components is a bunch of different technologies bundeled together allowing the user to create reusable custom elements with their functionalities, which later on can be re-used and utilized in your web apps.
```
Comment (rawdaGastan) [inline tutorial.md]: bundeled -> bundled

File: tutorial.md [hunk 8]
```diff
@@ -1,10 +1,168 @@
...
+- Web components are based on four main specifications:
+
+    1. Custom Elements
+
+        Custom Elements specifications lay the foundation for designing and using new types of DOM elements. According to it authors can define HTML tags with their behaviours and styles.
```
Comment (rawdaGastan) [inline tutorial.md]: behaviours -> behaviors

File: tutorial.md [hunk 9]
```diff
@@ -1,10 +1,168 @@
...
+    This will be the newly created tag:
+    ```bash
+    <cover-letter></cover-letter>
+
+    ```
 
 ### [Svelte](https://svelte.dev/)
 
-Svelte is a modern JavaScript framework used to build static web applications. Users use Svelte to build reusable components for any type of projects.
+Svelte is a modern JavaScript framework used to build fast web applications. Users use Svelte to either build the entire app woth it or build reusable components that can work as standalone packages for any type of projects.
```
Comment (rawdaGastan) [inline tutorial.md]: woth -> with

File: tutorial.md [hunk 10]
```diff
@@ -1,10 +1,168 @@
...
+    ![](img/hello-tf.png)
+
+Users sometimes need to access values by multiple unrelated components and that's where stores are used.
+A svelte store is an object with a subscribe method that allows interested parties to be notified whenever the store value changes.
+
+- Store Example
+
+    A count store example which includes increment, decrement and reset methods. In this example we're going to make three re-usable components Incrementer , Decrementer and Resetter.
+
+    In ```store.js``` file:
```
Comment (rawdaGastan) [inline tutorial.md]: store.js -> stores.js

File: tutorial.md [hunk 11]
```diff
@@ -1,10 +1,168 @@
...
+    </button>
+    ``` 
+
+    In ```App.svelte``` file:
+
+    ```bash
+    <script>
+        import { count } from './stores.js';
+
+        function decrement() {
```
Comment (rawdaGastan) [inline tutorial.md]: why adding 2 scripts? I think decrement() is not used

File: tutorial.md (hunk 3 above)
Comment (rawdaGastan) [inline tutorial.md]: reason -> reasons

File: tutorial.md [hunk 12]
```diff
@@ -14,7 +172,6 @@ One of the main reason svelte is used in developing our [weblets](https://play.g
 - Prepare the Flist
 - Create a a weblet
```
Comment (rawdaGastan) [inline tutorial.md]: Create a a weblet -> Create a weblet

File: tutorial.md [hunk 13]
```diff
@@ -42,7 +199,7 @@ In our case, Pasty already has a Dockerfile in its repository. We need to add so
 ```bash
 RUN mkdir ./file_name/
```
Comment (rawdaGastan) [inline tutorial.md:192]: I think we need to explain this more? where to put it in the docker?

File: tutorial.md [hunk 14]
```diff
@@ -62,11 +219,468 @@ docker push image-name
...
+
+When the image is ready, some information will be displayed.
+
+![](img/flist_ready.png)
+
+To Navigate to the created Flist Either search with the newly created file name in the search tab.
+
+![](img/search.png)
+
+Or Navigate to your repository in the contributers section from the Zero-Os Hub and navigate to the newly created Flist.
```
Comment (rawdaGastan) [inline tutorial.md]: contributers -> contributors

File: tutorial.md [hunk 15]
```diff
@@ -62,11 +219,468 @@ docker push image-name
...
+    grid3_client: GridClient;
+  }
+}
+```
+Next step is to install dependencies:
+```bash
+yarn
+```
+
+Now that the configuration is all set up, we need to prepare our classes and depoloyment files.
```
Comment (rawdaGastan) [inline tutorial.md]: depoloyment -> deployment

**Review by rawdaGastan — CHANGES_REQUESTED** [review]
simple comments :D

**Comment by rawdaGastan** [comment]
my comments are done 👍🏻

**Comment by xmonader** [comment]
great job!!!

**Pull request merged**